### ISSUE TITLE
EdgeProps markerEnd/markerStart wrong type

### ISSUE THREAD

**Issue by gpkc** [wbkd/react-flow#3149]
### Describe the Bug

`markerEnd` and `markerStart` are typed just as `string`:

https://github.com/wbkd/react-flow/blob/main/packages/core/src/types/edges.ts#L117

They should allow `EdgeMarker` as well (see e.g. https://reactflow.dev/docs/api/edges/base-edge/)

### Steps to Reproduce the Bug or Issue

1. Create a custom marker config
2. Pass it to e.g. `BaseEdge` `markerEnd` prop
3. Typescript will fail

### Expected behavior

As per the docs, it should work

### Platform

Platform not relevant

**Comment by bcakmakoglu** [comment]
There might be a misunderstanding about what the edge props `markerStart` and `markerEnd` are 😄 

When you define an edge you can pass a `string` or `EdgeMarker` type to `edge.markerStart` or `edge.markerEnd` as you have understood correctly I assume.
This will then be parsed and a marker will be created.
Afterwards an url for this marker will be generated.
The *url*, i.e. ``url(#${getMarkerId(markerStart, rfId)})`` for the marker will be passed to your custom edge as a prop `markerStart` or `markerEnd` so you can just directly use this url for a `path` element for example.

```html
<path :marker-end="markerEnd" ... `>
```

**Comment by moklick** [comment]
Yep, that's correct. At the very end (when we pass them to the path) markerStart and markerEnd are just strings. React Flow can create markers for you when you are passing them as edge options. You can also create your own markers without edge options. It's a standard SVG feature: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/marker-end

**Comment by gpkc** [comment]
@bcakmakoglu @moklick Sorry but I still don't get it. As per the docs for `BaseEdge`:
https://reactflow.dev/docs/api/edges/base-edge/

> markerStart: string or [marker config](https://reactflow.dev/docs/api/edges/edge-options/#edgemarker)
> markerEnd: string or [marker config](https://reactflow.dev/docs/api/edges/edge-options/#edgemarker)

For me, this would mean that I can pass a "marker config" i.e. an object with attributes defined [here](https://reactflow.dev/docs/api/edges/edge-options/#edgemarker) to the `markerStart` or `markerEnd` props of `BaseEdge`.

However, the PropTypes for `BaseEdge` from the code do not reflect that:
https://github.com/wbkd/react-flow/blob/main/packages/core/src/types/edges.ts#L122

**Comment by bcakmakoglu** [comment]
The docs seem wrong about this, the expected type for  the`BaseEdge` props `markerEnd` & `markerStart` should just be `string`.

You don't pass a marker config to the edge components but use it in the edge definition:
```ts
const initialEdges: Edge[] = [
  { id: "e1-2", source: "1", target: "2", markerEnd: { /* marker config */ },
  { id: "e1-3", source: "1", target: "3" }
];

// ...

  const [edges, setEdges, onEdgesChange] = useEdgesState(initialEdges);
```

As mentioned previously, these marker definitions are then read, markers created and then the "url" for that marker is passed to you.

**Comment by zurka1** [comment]
I'm also running into this issue! I'm trying to create a custom edge that will have the markerEnd defined with in the custom edge so users won't have to pass the markerEnd property. Is there no way to do this since the BaseEdge only accepts a string? 

Also here is the link to the docs that are incorrect. https://reactflow.dev/docs/api/edges/base-edge/

**Comment by moklick** [comment]
As @bcakmakoglu already said, the docs were wrong about this. I already [fixed it](http://localhost:3000/docs/api/edges/base-edge/#prop-types) and added a tiny section about the internal [marker handling](https://reactflow.dev/docs/api/edges/edge-options/#markers). 

Basically you have two options: 

1. use dynamically created arrows via markerStart/End edge options
2. create your own set of markers (this could also happen dynamically) and use those ids directly

**Comment by darrel1925** [comment]
@moklick this issue is not fixed on version `11.10.1`

TEMP SOLUTION:

To add `MarkerType.ArrowClosed` on my custom edge, I need to add the string "url(#1__color=black&type=arrowclosed)" to the `markerEnd` prop. Below is how you can an arrow to your custom edge

Example:
```
 <BaseEdge id={id} path={edgePath} markerEnd={"url(#1__color=black&type=arrowclosed)"}/>
```

**Comment by kevinschaich** [comment]
alternative b/c that ID seems pretty brittle

```typescript
import { BlockDependencyProps } from '@/lib/types'
import { BaseEdge, getStraightPath } from 'reactflow'

export default function CanvasBlockDependency(props: BlockDependencyProps) {
    const [edgePath] = getStraightPath({
        sourceX: props.sourceX,
        sourceY: props.sourceY,
        targetX: props.targetX,
        targetY: props.targetY,
    })
    return (
        <>
            <defs>
                <marker
                    id='triangle'
                    markerWidth='12'
                    markerHeight='12'
                    viewBox='-10 -10 20 20'
                    markerUnits='strokeWidth'
                    orient='auto-start-reverse'
                    refX='0'
                    refY='0'
                >
                    <polyline
                        stroke-linecap='round'
                        stroke-linejoin='round'
                        points='-5,-4 0,0 -5,4 -5,-4'
                        style={{ stroke: 'var(--primary)', fill: 'var(--primary)', strokeWidth: 1 }}
                    ></polyline>
                </marker>
            </defs>
            <BaseEdge
                id={props.id}
                path={edgePath}
                style={{ stroke: 'var(--primary)', strokeWidth: 2 }}
                markerEnd={'url(#triangle)'}
            />
        </>
    )
}
```
